builtin/aws/lambda/releaser.go

```diff
@@ -0,0 +1,323 @@
+		}
+	}
+	log.Info("Created permission", "permission", *p.Statement)
+
+	o, err := lambdasrv.CreateFunctionUrlConfig(&lambda.CreateFunctionUrlConfigInput{
+		// Todo: make AuthType configurable via HCL
+		AuthType:     aws.String(lambda.FunctionUrlAuthTypeNone),
+		FunctionName: a.AliasArn,
+		Cors:         &lambda.Cors{},
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			log.Error("Error creating function url config", "error", aerr.Code(), "message", aerr.Message())
+		}
+		return err
+	}
+	log.Info("Created function url config", "url", *o.FunctionUrl)
+
+	return nil
+}
+
+func (r *Releaser) resourceFunctionUrlDestroy(
+	ctx context.Context,
+	sess *session.Session,
+	sg terminal.StepGroup,
+) error {
+	step := sg.Add("Destroying Lambda URL...")
+	step.Update("Destroyed Lambda URL...")
+	step.Done()
+	return nil
+}
+
+func (r *Releaser) Release(
+	ctx context.Context,
+	log hclog.Logger,
+	src *component.Source,
+	ui terminal.UI,
+	// waypoint automatically injects the previous component's output
+	// - https://www.waypointproject.io/docs/extending-waypoint/passing-values
+	dep *Deployment,
+) (*Release, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	log.Info("Creating Function URL...")
+	log.Info("Deployment details", "deployment", dep)
+	log.Info("Deployment details", "FuncArn", dep.FuncArn, "VerArn", dep.VerArn)
+
+	// Create our resource manager and create
+	rm := r.resourceManager(log)
+	if err := rm.CreateAll(
+		ctx, log, sg, ui, src,
+		dep,
+	); err != nil {
+		log.Info("Error creating resources", "error", err)
+		return nil, err
+	}
+
+	// Get our function url state to verify
+	fnUrlState := rm.Resource("function_url").State().(*Resource_FunctionUrl)
+	log.Info("Function URL state", "url", fnUrlState.Url)
+	if fnUrlState == nil {
+		return nil, status.Errorf(codes.Internal, "function url state is nil, this should never happen")
+	}
+
+	return &Release{
+		Url:     fnUrlState.Url,
+		FuncArn: dep.FuncArn,
+		VerArn:  dep.VerArn,
+	}, nil
+}
+
+// Destroy will modify or delete Listeners, so that the platform can destroy the
+// target groups
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	rm := r.resourceManager(log)
+
+	// Destroy All
+	return rm.DestroyAll(ctx, log, sg, ui, sess)
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// "AWS_IAM" or "NONE"
+	AuthType string `hcl:"auth_type,optional"`
+	// todo(kevinwang): CORS
+}
+
+func (r *Releaser) Status(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) (*sdk.StatusReport, error) {
+	var report sdk.StatusReport
+	report.External = true
+	defer func() {
+		report.GeneratedTime = timestamppb.Now()
+	}()
```

Comment:
This would be a good thing to double check on - but I believe the SDK will generate the time for you. But I'm not sure!

Reply:
I copied the `Status` implementation from [`aws-alb`](https://github.com/hashicorp/waypoint/blob/6dd2723d0dd4790eafb27facd6e94ed2f2a18cfa/builtin/aws/alb/releaser.go#L782) 😅 , and also hadn't yet read through the [docs](https://www.waypointproject.io/docs/extending-waypoint/plugin-interfaces/status).

It looks like several other built-in plugins explicitly pass `timestamppb.Now(),` to the status report
- [`nomad/platform`](https://github.com/hashicorp/waypoint/blob/6dd2723d0dd4790eafb27facd6e94ed2f2a18cfa/builtin/nomad/platform.go#L371)
- [`k8s/releaser`](https://github.com/hashicorp/waypoint/blob/0119dccca2a19a0b242d8d2db9b7aa70d2e38069/builtin/k8s/releaser.go#L856)
- [`nomad/jobspec/releaser`](https://github.com/hashicorp/waypoint/blob/b6e78045a94d2ca7ab5703b1ee35405fd0c25be3/builtin/nomad/jobspec/releaser.go#L418)

I'm also looking at [`waypoint-plug-sdk`](https://github.com/hashicorp/waypoint-plugin-sdk/blob/main/framework/resource/manager.go#L446) code for the first time, and not entirely sure if this line means that there is a default value for `GeneratedTime ` 🤔

Reply:
Yep totally.... I believe we added automatic timestamp generation in the SDK after those plugins were written so I think it's not required anymore? https://github.com/hashicorp/waypoint-plugin-sdk/blob/3c2ebd53709006279ffd89272b6e4cbb76e71228/framework/resource/manager.go#L444-L451

Those other plugins might need to be updated. I'd give it a shot and check the reported generated time to confirm it works. If so we could make a PR to fix up those other plugins!

Reply:
Yea according the UI, `GeneratedTime` looks like it doesn't need to be set. I'll go ahead and remove it here
<img width="678" alt="image" src="https://user-images.githubusercontent.com/26389321/163076965-bcb2ea0b-e047-491d-bf8a-5f79f973d785.png">

Reply:
removed: https://github.com/hashicorp/waypoint/pull/3187/commits/0bda64cd1d5f9e95756d12a1fb013efd337b7552